src/PowerShellEditorServices.Hosting/EditorServicesRunner.cs

```diff
@@ -0,0 +1,276 @@
+            //  - Start the LSP server
+            //  - Possibly start the debug server
+            //  - Wait for the LSP server to finish
+
+            // Unsubscribe the host logger here so that the integrated console is not polluted with input after the first prompt
+            _logger.Log(PsesLogLevel.Verbose, "Starting server, deregistering host logger and registering shutdown listener");
+            foreach (IDisposable loggerToUnsubscribe in _loggersToUnsubscribe)
+            {
+                loggerToUnsubscribe.Dispose();
+            }
+
+            // Write the integrated console banner
+            _config.PSHost.UI.WriteLine("\n=== PowerShell Integrated Console ===");
+
+            PsesLanguageServer languageServer = await CreateLanguageServer(hostStartupInfo).ConfigureAwait(false);
+
+            Task<PsesDebugServer> debugServerCreation = null;
+            if (creatingDebugServer)
+            {
+                debugServerCreation = CreateDebugServerWithLanguageServer(languageServer);
+            }
+
+            languageServer.StartAsync();
+
+            if (creatingDebugServer)
+            {
+                StartDebugServer(debugServerCreation);
+            }
+
+            await languageServer.WaitForShutdown().ConfigureAwait(false);
+
+            // Resubscribe host logger to log shutdown events to the console
+            _logger.Subscribe(new PSHostLogger(_config.PSHost.UI));
+        }
+
+        private async Task RunTempDebugSession(HostStartupInfo hostDetails)
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Running temp debug session");
+            PsesDebugServer debugServer = await CreateDebugServerForTempSession(hostDetails).ConfigureAwait(false);
+            _logger.Log(PsesLogLevel.Verbose, "Debug server created");
+            await debugServer.StartAsync().ConfigureAwait(false);
+            _logger.Log(PsesLogLevel.Verbose, "Debug server started");
+            await debugServer.WaitForShutdown().ConfigureAwait(false);
+            return;
+        }
+
+        private async Task StartDebugServer(Task<PsesDebugServer> debugServerCreation)
+        {
+            PsesDebugServer debugServer = await debugServerCreation.ConfigureAwait(false);
+
+            // When the debug server shuts down, we want it to automatically restart
+            // To do this, we set an event to allow it to create a new debug server as its session ends
+            if (!_alreadySubscribedDebug)
+            {
+                _logger.Log(PsesLogLevel.Diagnostic, "Subscribing debug server for session ended event");
+                _alreadySubscribedDebug = true;
+                debugServer.SessionEnded += DebugServer_OnSessionEnded;
+            }
+
+            _logger.Log(PsesLogLevel.Diagnostic, "Starting debug server");
+            debugServer.StartAsync();
+            return;
+        }
+
+        private Task RestartDebugServer(PsesDebugServer debugServer)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)